rllib/offline/offline_prelearner.py

```diff
@@ -240,17 +267,39 @@ def _map_to_episodes(
                 episode = SingleAgentEpisode(
                     id_=batch[schema[Columns.EPS_ID]][i],
                     agent_id=agent_id,
+                    # Observations might be (a) serialized and/or (b) converted
+                    # to a JSONable (when a composite space was used). We unserialize
+                    # and then reconvert from JSONable to space sample.
                     observations=[
-                        unpack_if_needed(obs),
-                        unpack_if_needed(batch[schema[Columns.NEXT_OBS]][i]),
+                        convert(unpack_if_needed(obs), observation_space)
```

Comment:
What if `observation_space` is None here? Could that happen? Or should we make the arg non-optional?

Reply:
This could happen and is in fact okay. As it uses `from_jsonable_if_needed` the conversion simply does not take place, if the input space is `None`. We need them only, if we want to convert from JSONable data types to a composite space type.